Sources/LeafKit/LeafSerializer.swift

```diff
@@ -137,8 +137,9 @@ struct LeafSerializer {
         for (idx, item) in array.enumerated() {
             var innerContext = self.data
             
-            if idx == 0 { innerContext["isFirst"] = .bool(true) }
-            else if idx == array.count - 1 { innerContext["isLast"] = .bool(true) }
+            innerContext["isFirst"] = .bool(idx == 0)
+            innerContext["isLast"] = .bool(idx == array.count - 1)
```

Comment:
Likewise here, use `idx == array.index(before: array.endIndex)`. Don't worry, in release builds the compiler will inline it all the way back down to `idx == endIndex - 1` 🙂 (And in this form it may even successfully peephole/CSE the subtraction expression outside the loop, since `endIndex` is less complex than `count` in the general case.. but I digress.)

Reply:
Updated